src/acl_mem.cpp

```diff
@@ -2771,15 +2775,18 @@ CL_API_ENTRY void *CL_API_CALL clEnqueueMapImageIntelFPGA(
     if (image->mem_object_type == CL_MEM_OBJECT_IMAGE2D ||
         image->mem_object_type == CL_MEM_OBJECT_IMAGE1D ||
         image->mem_object_type == CL_MEM_OBJECT_IMAGE1D_BUFFER) {
-      if (image_slice_pitch != NULL) {
-        *image_slice_pitch = 0;
+      if (image_slice_pitch == NULL) {
+        BAIL_INFO(CL_INVALID_VALUE, command_queue->context,
+                  "Invalid slice pitch provided");
       }
```

Comment:
Yep I think I understand why the code is written this way. It is meant to change the value `image_slice_pitch` is pointing to as well since it is an input value to the function. Therefore, it is allowed to have `NULL` for certain `image_types`, which indicates that there's nothing to be performed on the pointer `image_slice_pitch`. 

I implemented the requested changes. Coverity is happy!